internal/packager/registry.go

```diff
@@ -95,14 +95,8 @@ func Push(app *types.App, namespace, tag, repo string) (string, error) {
 }
 
 func createImageName(app *types.App, namespace, tag, repo string) string {
```

Comment:
namespace / tag parameters should be dropped, and repo should be replaced by registryReference.
Thus, if registryReference is not set, we generated the ref trough app metadata, but if it is set, we consider it is full (and don't try to combine stuff from user input with things defined in the metadata, it is too confusing).
So if I have 
```
metadata.yml:
name: simon/toto
version: 0.42
```
if I do `docker app push`, t will infer the ref to be simon/toto:0.42
But if I use an explicit reference like `docker app push -t myteam/myapp`, it will push to `myteam/myapp:latest`.
If I use a tagg-qualified reference `docker app push -t myteam/myapp:production` it will push to `myteam/myapp:production`.

I find it much clearer: 
- you don't give a tag -> we try to generate something meaningfull
- you give us a tag -> we use it without messing with it whatsoever

Reply:
+1 on this suggestion. I've had similar in passing vague thoughts but this codifies them into something concrete.